### ISSUE TITLE
`null` value annotation, when referecing Android's `BuildConfig.`

### ISSUE THREAD

**Issue by gmazzo** [google/ksp#350]
We have a custom processor for our FeatureToggles solution:
```kotlin
@FeatureToggle(defaultsTo = BuildConfig.HTTP_REQUEST_DEBUGGING)
```

With `kapt`, the value correctly read from Android's `BuildConfig` class (generated by `buildConfigField`).
With `ksp`, we get `null` instead on `KSValueArgument[name = 'defaultsTo'].value`.

I've seen a similar issue when there are compilations issues (like a missing class). I've not confirmed it, but I'm pretty sure that generated code from `generate${variant}BuildConfig` task's output is getting added as classpath for `ksp` task.

**Comment by elihart** [comment]
Is this fixed by https://github.com/google/ksp/pull/606 ?

**Comment by gmazzo** [comment]
Nope, I just re-tried and the whole `defaultsTo` argument is missing:
![image](https://user-images.githubusercontent.com/513566/133400632-0909b2ab-b818-4fcd-8bc8-303d56a56a10.png)

**Comment by bomiyr** [comment]
Any news about this issue? I still can reproduce it with lates ksp version (1.7.20)
If it helps, I created repository reproducing this issue: https://github.com/bomiyr/KspBug

**Comment by danysantiago** [comment]
This issue is still present and affects some Room users, see: https://issuetracker.google.com/245793067

Specifically when using `BuildConfig` as the database version:
```
@Database(
    entities = [MyEntity::class],
    version = BuildConfig.VERSION_CODE
    exportSchema = false
)
```
compilation will fail with Room due to the `KSAnnotation` missing the `version` in its argument list because `BuildConfig.VERSION_CODE` was not resolved which ultimately leads to Room's KSP processor crashing as it always expects a value.

Note that a side issue is that KSP seems to omit primitive arguments from the annotation when the argument comes from an unresolved constant in another class, this is somewhat similar to https://github.com/google/ksp/issues/1474.

**Comment by AmrAfifiy** [comment]
We're working on a proper fix for the issue by exposing an API for generated java sources by the android gradle plugin to be consumed by KSP gradle plugin. Meanwhile, you can add the following snippet to your build file as a workaround to wire build config generated classes to the ksp task.

```
 androidComponents {
    onVariants(selector().all(), { variant ->
        afterEvaluate {
            // This is a workaround for https://issuetracker.google.com/301244513 which depends on internal
            // implementations of the android gradle plugin and the ksp gradle plugin which might change in the future
            // in an unpredictable way.
            project.tasks.getByName("ksp" + variant.name.capitalize() + "Kotlin") {
                def buildConfigTask = (com.android.build.gradle.tasks.GenerateBuildConfig) project.tasks.getByName("generate" + variant.name.capitalize() + "BuildConfig")

                ((org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompileTool) it).setSource(
                        buildConfigTask.sourceOutputDir
                )
            }
        }
    })
}
```

**Comment by hfmehmed** [comment]
I believe this is a long fixed issue and closing based on that. If there is still problems, please create a new issue with reproducing project
